scripts/vaccination_alerts/send-emails.ts

```diff
@@ -0,0 +1,206 @@
+    if (dryRun) {
+      this.updateSentEmailCounters(email, alert);
+      return;
+    }
+
+    // @ts-ignore
+    let sendData: EmailSendData = null; //  generateAlertEmailData(email, locationAlert);
+    try {
+      await this.emailService.sendEmail(sendData);
+      await this.onEmailSent(email, alert);
+    } catch (err) {
+      if (isInvalidEmailError(err)) {
+        await this.onInvalidEmail(email);
+      } else {
+        console.error(
+          `Error sending email ${email}, ${alert.fips} - ${alert.emailAlertVersion}.`,
+          err,
+        );
+        this.errorCount += 1;
+      }
+    }
+  }
+}
+
+/**
+ * Given a list of users to email and a list of locations, email those
+ * users with alert information about those locations.
+ *
+ * Takes in a file with {[fips: string]: {...data}} and file like {[fips: string]: [emails]}
+ *
+ * Also takes in a parameter to actual send the email. Otherwise will just generate the files
+ *
+ * You can run via `yarn send-emails fipsToAlertFilename currentSnapshot [send]`
+ */
+async function main(
+  alertPath: string,
+  batchSize: number,
+  dryRun: boolean,
+  singleEmail?: string,
+  fips?: string,
+) {
+  const alertsByFips = readVaccinationAlerts(alertPath);
+  const alerts = _.values(alertsByFips);
+
+  const emailService = new EmailService();
+  const db = getFirestore();
+
+  const alertEmailService = new SendVaccinationAlertsService(db, emailService);
+
+  const emailAlertTuples = _.flatten(
+    await Promise.all(
+      alerts.map(async alert => {
+        const emails = await alertEmailService.fetchEmailsForAlert(alert);
+        return (
+          emails
+            // if an email is specified, filter out all other emails. Used for testing
+            .filter(email => (singleEmail ? email === singleEmail : true))
+            .map(email => ({ email, alert }))
+        );
+      }),
+    ),
+  );
+
+  const emailBatches = _.chunk(emailAlertTuples, BATCH_SIZE);
+  console.log(
+    `Sending ${emailAlertTuples.length} emails/${emailBatches.length} batches.`,
+  );
+  for (const batch of emailBatches) {
+    await Promise.all(
+      batch.map((data: { email: string; alert: RegionVaccinePhaseInfo }) =>
+        alertEmailService.sendAlertEmail(data.email, data.alert, dryRun),
+      ),
+    );
+  }
+  const emailsSent = alertEmailService.emailSent;
+  const errorCount = alertEmailService.errorCount;
+  const uniqueEmailAddresses = alertEmailService.uniqueEmailAddress;
+  const invalidEmailAddresses = alertEmailService.invalidEmailCount;
+  console.info(`Total emails to be sent: ${emailsSent}.`);
+  console.info(
+    `Unique Email addresses: ${Object.keys(uniqueEmailAddresses).length}.`,
+  );
+  console.info(`Invalid emails removed: ${invalidEmailAddresses}.`);
+
+  if (!dryRun) {
+    // setLastSnapshotNumber(db, `${currentSnapshot}`);
+    // If we aren't in a dry run but there's clearly some non trivial errors exit with error
+    if (emailsSent < 1 || errorCount > 1) {
```

Comment:
I think that we need to make this script a bit more forgiving, there are some invalid emails in our Firebase collection (you can check the logs of the risk level alerts to see some of them).

Reply:
ah yeah i wasn't 100% sure what to do here

Reply:
this is actually okay, because we don't throw on invalid emails :)